### PR TITLE
chore(html): split html up in seperate files

### PR DESCRIPTION
**Pull request by evenstensberg**

### CODE REVIEW

File: html.mjs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+ */
+
+
+/**
+ * Creates a link with report results.
+ * @param {string} url URL to the result.
+ * @return {string} A link to the results.
+*/
+export const getResultsLink = (url) => (`
+      <p class="reportlink">
```
Comment (ebidel) [inline html.mjs]: 2 space indents

File: html.mjs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+ * limitations under the License.
+ */
+
+
+/**
+ * Creates a link with report results.
+ * @param {string} url URL to the result.
+ * @return {string} A link to the results.
+*/
+export const getResultsLink = (url) => (`
```
Comment (ebidel) [inline html.mjs]: let's use regular functions here. I don't think the arrow functions and lack of `return` gives us much.

**Review by ebidel — CHANGES_REQUESTED** [review]

**Comment by ev1stensberg** [comment]
Fixed now, let me know if the indentation is still off, literal indents are hard to fix 😆

File: server.mjs [hunk 3]
```diff
@@ -32,6 +32,8 @@ import * as WPTTool from './tools/wpt.mjs';
 import * as PSITool from './tools/psi.mjs';
 /* eslint-enable no-unused-vars */
 
+import {getResultsLink, createHTMLTemplate, createHTMLPage} from './html';
```
Comment (ebidel) [inline server.mjs]: `from './html.mjs';`

File: tools/wpt.mjs [hunk 4]
```diff
@@ -19,7 +19,8 @@ const {URL} = url;
 import fetch from 'node-fetch';
 import * as tools from '../public/tools.mjs';
 
-const API_KEY = '7dce091be3b64d95b7812bb5211ad5a9';//'A.04c7244ba25a5d6d717b0343a821aa59';
+// 'A.04c7244ba25a5d6d717b0343a821aa59';
```
Comment (ebidel) [inline tools/wpt.mjs]: please revert

File: tools/tms.mjs [hunk 5]
```diff
@@ -55,7 +55,9 @@ async function run(browser, url) {
 
   // await page.goto(TOOL.url);
   // await page.waitForSelector(TOOL.urlInputSelector);
-  await page.goto(`${TOOL.url}?url=${url}`);//, {waitUntil: 'networkidle2'});
+
+  // , {waitUntil: 'networkidle2'});
```
Comment (ebidel) [inline tools/tms.mjs]: please revert

**Review by ebidel — COMMENTED** [review]
Can we address the other comments too?

**Comment by ev1stensberg** [comment]
Changed what you requested. Want me to rebase into one comment as well?

**Comment by ev1stensberg** [comment]
@ebidel anything more that needs attention before this is ready?

File: html.mjs [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+
+/**
+ * Creates a link with report results.
+ * @param {string} url URL to the result.
+ * @return {string} A link to the results.
+*/
+export function getResultsLink(url) {
+  return (
+    `<p class="reportlink">
```
Comment (ebidel) [inline html.mjs]: you don't need the `()` around these.

File: html.mjs [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        Results available at: <a href="${url}" target="_blank">${url}</a>
+    </p>`
+  );
+}
+
+/**
+ * Creates a html template with meta-data.
+ * @param {string} name Name of the  test.
```
Comment (ebidel) [inline html.mjs]: `@param {!{name: string, desc: string, url: string}}` with the descriptions you have.

File: html.mjs [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+ * @param {string} screenshot Screenshot and result of the test.
+ * @return {string} A html template with meta-data.
+ */
+export function createHTMLTemplate({name, desc, url}, resultsLink, screenshot) {
+  return (
+    `<h3 class="title">${name} results</h3>
+        <div>
+            <div class="desc">
+            About this tool: ${desc}
+            Learn more at <a href="${url}" target="_blank">${url}</a>
```
Comment (ebidel) [inline html.mjs]: indentation is weird throughout his html block

File: html.mjs [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+            }
+            h1, h2, h3, h4 {
+                font-weight: inherit;
+                margin: 0;
+            }
+            h1 {
+                color: var(--purple);
+            }
+            a {
+                color: var(--purple);
```
Comment (ebidel) [inline html.mjs]: 2 space indents

File: html.mjs [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+                <h4 class="date">Google I/O 2018 &middot; ${(new Date()).toLocaleDateString()}</h4>
+            </div>
+            <div>
+                <img src="https://storage.googleapis.com/io-2018.appspot.com/v1/hashtag.gif">
+            </div>
+            </header>
+            ${content}
+        </body>
+    </html>`
+  );
```
Comment (ebidel) [inline html.mjs]: remove the `()`

**Review by ebidel — CHANGES_REQUESTED** [review]

File: html.mjs (hunk 10 above)
Comment (ev1stensberg) [reply html.mjs]: return statement?

**Comment by ev1stensberg** [comment]
@ebidel PTAL

File: html.mjs (hunk 10 above)
Comment (ebidel) [reply html.mjs]: Yes. You can return the template literals. No need to wrap in `()`.

File: html.mjs [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+
+/**
+ * Creates the html page based on yielded results.
+ * @param {string} content Body content and results to be injected.
+ * @return {string} Created html page.
+ */
+export function createHTMLPage(content) {
+  return (
+    `<html>
+        <head>
```
Comment (ebidel) [inline html.mjs]: can you keep the 2-space like before?

File: html.mjs [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+ * @param {Object} object with Name, Description and url.
+ * @param {string} resultsLink Link to the result.
+ * @param {string} screenshot Screenshot and result of the test.
+ * @return {string} A html template with meta-data.
+ */
+export function createHTMLTemplate({name, desc, url}, resultsLink, screenshot) {
+  return (
+    `<h3 class="title">${name} results</h3>
+     <div>
+        <div class="desc">
```
Comment (ebidel) [inline html.mjs]: keep 2 space indentation like before.

**Review by ebidel — CHANGES_REQUESTED** [review]

**Comment by ev1stensberg** [comment]
Changed now @ebidel . `createHTMLTemplate` was a bit tricky to indent has the h3 is one liner

**Pull request closed without merging**